benchmark.go

```diff
@@ -126,15 +127,30 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, m benchmarkMe
 		out.Fatalf("Failed to warmup connections for benchmark: %v", err)
 	}
 
-	statter, err := statsd.NewClient(logger, opts.StatsdHostPort, allOpts.TOpts.ServiceName, allOpts.ROpts.Procedure)
+	globalStatter, err := statsd.NewClient(logger, opts.StatsdHostPort, allOpts.TOpts.ServiceName, allOpts.ROpts.Procedure)
 	if err != nil {
 		out.Fatalf("Failed to create statsd client for benchmark: %v", err)
 	}
 
 	var wg sync.WaitGroup
 	states := make([]*benchmarkState, len(connections)*opts.Concurrency)
-	for i := range states {
-		states[i] = newBenchmarkState(statter)
+
+	for i, c := range connections {
+		statter := globalStatter
+
+		if opts.PerPeerStats {
+			// If per-peer stats are enabled, dual emit metrics to the original value
+			// and the per-peer value.
+			prefix := fmt.Sprintf("peer.%v.", c.peerID)
+			statter = statsd.MultiClient{
+				globalStatter,
+				statsd.NewPrefixedClient(globalStatter, prefix),
+			}
+		}
+
+		for j := 0; j < opts.Concurrency; j++ {
+			states[i*opts.Concurrency+j] = newBenchmarkState(statter)
```

Comment:
Agree that the calculation is odd, but we have to ensure that the indexes here align with how the goroutines are started, otherwise, where we do:
```
	for i, c := range connections {
		for j := 0; j < opts.Concurrency; j++ {
			state := states[i*opts.Concurrency+j]
```

Using the same thing here ensures we're doing the right thing. If we just append, I think there's a higher risk we get it wrong